private/bufpkg/bufconnect/errors.go

```diff
@@ -47,3 +47,31 @@ func AsAuthError(err error) (*AuthError, bool) {
 	ok := errors.As(err, &authErr)
 	return authErr, ok
 }
+
+// AugmentedConnectError wraps an error, providing the Procedure and Addr of
+// the request for introspection in callers.
+type AugmentedConnectError struct {
+	cause     error
```

Comment:
Could be left as a *connect.Error.

Reply:
I [wanted to use the original error](https://github.com/bufbuild/buf/blob/addef8cd746cd70bb9540adbd57018eb18021009/private/bufpkg/bufconnect/interceptors.go#L39-L42), which I suppose might not necessarily _immediately_ be a *connect.Error, in the case that an interceptor further down the chain replaced it with a newly wrapped error. Not sure if that's overthinking it, though.